python/tests/test_types.py

```diff
@@ -161,3 +187,25 @@ def test_nested_field():
     assert field_var.field_id == 1
     assert isinstance(field_var.type, StructType)
     assert str(field_var) == str(eval(repr(field_var)))
+
+
+@pytest.mark.parametrize(
+    "input_type,equal_object,not_equal_object",
+    [
+        (BooleanType, BooleanType(), IntegerType()),
+        (IntegerType, IntegerType(), BooleanType()),
+        (LongType, LongType(), DoubleType()),
+        (FloatType, FloatType(), IntegerType()),
+        (DoubleType, DoubleType(), FloatType()),
+        (DateType, DateType(), TimeType()),
+        (TimeType, TimeType(), TimestampType()),
+        (TimestampType, TimestampType(), TimestamptzType()),
+        (TimestamptzType, TimestamptzType(), TimestampType()),
+        (StringType, StringType(), UUIDType()),
+        (UUIDType, UUIDType(), StringType()),
+        (BinaryType, BinaryType(), FixedType(6)),
+    ],
+)
+def test_primitive_equality(input_type, equal_object, not_equal_object):
+    assert equal_object == input_type()
+    assert not_equal_object != input_type()
```

Comment:
I doubt this will change much but for the negative case you can use a [product](https://docs.python.org/3/library/itertools.html#itertools.product) and filter out objects of the same type to do an exhaustive test.

Reply:
I think the problem of doing it that way is that you end up using equality to build the test cases for inequality. So by filtering the list you're testing that objects are unequal if they were not filtered because they were equal.

Reply:
One could use index in the list equality check to avoid this problem

Reply:
Let me give it a try. Will update it very soon.

Reply:
Updated by using pytest features.